aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -113,7 +115,13 @@ public void initialize(Map<String, String> properties) {
 
     // Do not override s3 client if it was provided
     if (s3 == null) {
-      this.s3 = AwsClientFactories.from(properties)::s3;
+      if (PropertyUtil.propertyAsBoolean(properties, AwsProperties.LAKE_FORMATION_ENABLED, false)) {
```

Comment:
I think it makes more sense to move this logic to `AwsClientFactories.from` instead of `S3FileIO.initialize`, because S3 does not really care about LakeFormation. 

And we should throw an exception when a different client factory value is set for now. We could extend this in the future to allow more customization.